packages/plugin-tools/src/editors/ResourcesList.tsx

```diff
@@ -135,7 +136,10 @@ export function PickResourceList(props: PickResourceListProps) {
                 omitIds={store.value}
                 onChange={action(resource => {
                   if (resource) {
-                    props.onChange([...(store.value || []), String(resource.id)]);
+                    if(store.value){
+                      store.value.push(String(resource.id));
+                    }
+                    props.onChange(store.value);
```

Comment:
The previous code returned `String(resource.id)` with or without there being value from prop. Now we are adding it conditionally. Also, it feels like store.value will mostly _always_ be `true`.
If we know we always get an Array in the `value` property then the check is unnecessary. If it can be anything other than Array, then I would update the code to 
```
if (!Array.isArray(store.value)) {
  store.value = [];
}

store.value.push(String(resource.id));
props.onChange(store.value);

```